packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -252,21 +253,25 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 
 		connection.binaryType = 'arraybuffer';
 
-		connection.onmessage = (event) => {
-			void this.onMessage(event.data, event.data instanceof ArrayBuffer);
-		};
+		connection.on('message', (event) => {
```

Comment:
The reason we moved away from this pattern is compatibility with deno and buns websocket. Unless they randomly started supporting EE-style ws, this will not work

Reply:
Simply can't do then. We need `connection.on('unexpected-response')`, there's no other way to fix this issue other than legitimately parsing `error.message` for an HTTP status code.

Reply:
Doesn't the error emitted onerror contain the http code? Also from the logs from the original issue, the gw connection closed with a 1006 shortly after the error, which should've retried, no?

Reply:
Looking through docs, Both seem to support `.addEventListener` (I think that doesn't also imply `.on`?). Can't tell if either support `unexpected-response`, though..

Reply:
> Doesn't the error emitted onerror contain the http code? Also from the logs from the original issue, the gw connection closed with a 1006 shortly after the error, which should've retried, no?

No, by [design even](https://stackoverflow.com/questions/21762596/how-to-read-status-code-from-rejected-websocket-opening-handshake-with-javascrip), because of web standards. And also no, because we fired an error event during initial `connect`. That's the point of the other code at the top of `onError`, to prevent firing it for recoverable I/O or network errors.